examples/dev_your_plugin/go.mod

```diff
@@ -17,10 +17,10 @@ module mosn.io/htnn/dev_your_plugin
 go 1.21
 
 require (
-	github.com/envoyproxy/envoy v1.29.4
+	github.com/envoyproxy/envoy v1.31.0
 	github.com/stretchr/testify v1.9.0
-	google.golang.org/protobuf v1.33.0
-	mosn.io/htnn/api v0.3.2
+	google.golang.org/protobuf v1.34.1
+	mosn.io/htnn/api v0.3.3-0.20240924080219-d01ea9d6301e
```

Comment:
Does it fair to say this api is not compatible for envoy 1.29 after the lib upgraded? Or we support both 1.28 and 1.31?

Reply:
I wrote a compatible layer for Envoy 1.29, so both Envoy 1.29 and 1.31 are supported. If we need to support Envoy 1.30, we can write a compatible layer for it.

Here are the steps to support Envoy 1.29: https://github.com/mosn/htnn/blob/main/site/content/en/docs/developer-guide/dataplane_support.md#choosing-the-target-data-plane-version. We use `replace in go.mod` and build tag to switch the Envoy SDK we used as Go doesn't support multiple version of the same library in one module. If you use the latest Envoy (1.31), you don't need to do anything - the support is out of the box.

Reply:
Thanks a lot! We're using envoy 1.31 for now, and having some old versions running in production as well. In case we will probably need to support multiple envoy versions in the long term for some useful filters, I would like to have your clarification here in the examples to guide us about maintaining multiple envoy versions for this example golang filter.

Reply:
@wonderflow 
Feel free to open an issue to ask for a new example about running HTNN on the old Envoy.
For now, you can refer to https://github.com/mosn/htnn/blob/d01ea9d6301ee5849fa9a927ffae3b4af38f21f3/manifests/images/dp/Dockerfile#L37-L42

In this living code we build a shared lib target to the istio 1.21.3's data plane (which is covered by the e2e tests).

BTW, I don't recommend using HTNN on Envoy < 1.29. Versions before that are too old unless you can backport the latest Envoy Golang filter to an old version.

Reply:
The dockefile above is helpful, thanks! What is our compatible strategy? I don't want old version support, but in the feature let's say when envoy 1.32 or newer version come out, when will we retire the 1.29 support?

Reply:
> What is our compatible strategy?

We will maintain the Envoy X support as long as Envoy still provides security updates for version X or the istio version used by Ant Group (currently 1.21) requires version X.

Reply:
that's cool